tests/k8s/conftest.py

```diff
@@ -0,0 +1,28 @@
+import pytest
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture(autouse=True)
+def check_k8s_vms(k8shosts):
+    """
+    This fixture runs before each k8s test to make sure that Kubernetes API server is reachable on all backend master servers
+    This fixture also runs after each test to start kubelet if necessary- in case test code stopped kubelet, and test failed before reaching logic to start kubelet
+
+    Args:
+    k8shosts:  Shortcut fixture for getting Kubernetes hosts
+    """
+    for i in range(1, len(k8shosts)):
```

Comment:
>1 [](start = 19, length = 1)

Start from 0?

Reply:
I need 1, 2, and 3. `len(k8shosts) = 4` because it includes HAProxy, but I just need m1, m2, and m3 here
```
for i in range(1, len(k8shosts)):
        k8shost = k8shosts['m{}'.format(i)]['host']
```

Reply:
Can you consider directly looping the dict?
```
for hostname, k8smaster in k8shosts.items():
```

Reply:
If I directly loop the dict, I would need to parse the hostname and see if it ends in `ha` to avoid running `check_master_ready()` on the HAProxy machine. Would this be advisable over the current setup?

I could save `k8shosts[hostname]` but this would make it more complicated get the VIP. Currently, I can get the VIP by retrieving `k8shosts[ha]`

Reply:
I changed it to directly loop through `backend_masters` in `K8sMasterCluster` object